pkg/server/server.go

```diff
@@ -116,7 +129,15 @@ func (s *Server) Delete(ctx context.Context, req *pbzk.DeleteRequest) (*pbzk.Del
 	if len(node.Children) > 0 {
 		return nil, fmt.Errorf("the node specified has children. Only leaf nodes can be deleted")
 	}
+	// Delete the actual node from the tree.
 	delete(parent.Children, nameToDelete)
+	// Clean up any references if this was ephemeral.
+	if node.NodeType == znode.ZNodeType_EPHEMERAL {
+		// If this session hasn't already been deleted, then clean up the reference to this node here.
+		if sess, ok := s.sessions[node.Creator]; ok {
+			delete(sess.EphemeralNodes, req.GetPath())
+		}
```

Comment:
Based on how we've written the CloseConnection() logic, this should never happen. Do we still want this here?